### ISSUE TITLE
What is the power consumption in sleep mode?

### ISSUE THREAD

**Issue by harmverbeek** [cnlohr/esplocalizer#7]
Is it < 20 uA?
That would make it a really useful IoT device.

**Comment by con-f-use** [comment]
There are different sleep modes in the esp8266, but yes, you can get in the ballpark of that but about everything on the chip will be disabled.

**Comment by harmverbeek** [comment]
how about the usb uart chip and the leds?
are they also powered down?

**Comment by con-f-use** [comment]
LEDs can be switched of in software or, in case of the power indicator LED desoldered on most ESP8266 modules - not all are the same. I don't know about the UART. Consult the data sheet and schematics for that.

**Comment by cnlohr** [comment]
THIS REMINDS ME!  I got permission to post a study I did on the sleeping options for the ESP on the esplocalizer!  It's really rough and totally not publish ready but here is a link to the google drive and some of the code I used.

https://drive.google.com/open?id=0B-rO-WAvEtVBU1FSOVRKZ3Npc1U

TL;DR:
 * Deep sleep on esplocaizer is 0.9mA
 * Sleeping, waiting for a GPIO wakeup is 1.8mA
 * Disconnected-from-wifi @pvvx light-sleep or as I call it "crazy sleep" where the processor is still running but not very utilized is 3.5-5.5mA depending on work load.  Thanks @pvvx you are one awesome dude.
 * Wifi on, light sleep ~12mA
 * Full bore: 72mA

**Comment by cnlohr** [comment]
The real killer for suuuper tiny power consumption on this board is a combination of the regulators, and static uses.  I don't actually know how much the LEDs use, but it's not much.

**Comment by harmverbeek** [comment]
0.9mA is 90 times too much for a battery powered sensor ;)
That's the problem with all sensor boards that include a USB UART chip, regulators and leds.
You have got to find a way to at least switch off the UART chip when not used.

**Comment by cnlohr** [comment]
This board does not include the uart chip on it.  USB on here is strictly for power.  It's everything else on-board, including the lithium ion charging circuit.
